### PR TITLE
Mm 16623 fix all initialism errors in channels/app and channels/api4 dir

### PR DESCRIPTION
**Pull request by TealWater**
#### Summary

Fixed all initialism errors in 14 files in accordance with the golang style doc found [here](https://github.com/golang/go/wiki/CodeReviewComments#initialisms).

List of warnings fixed:

variable naming conventions --> abbreviations like ID need to be capitalized
```
var-naming: var user_ID should be useID (revive)
var-naming: var teanId should be teamID (revive)
```
#### Ticket Link

Fix for: https://github.com/mattermost/mattermost/issues/16623

#### Note
**It may be wise to drop the first commit before merging** 
[088dd00](https://github.com/mattermost/mattermost/pull/25743/commits/088dd00a848c2c31952c29f0a2a28a9fb47fc3f7)
#### Release Note

```release-note
none
```

### CODE REVIEW

File: server/go.mod [hunk 1]
```diff
@@ -73,7 +73,7 @@ require (
 	golang.org/x/image v0.8.0
 	golang.org/x/net v0.17.0
 	golang.org/x/sync v0.3.0
-	golang.org/x/term v0.13.0
+	golang.org/x/term v0.15.0
```
Comment (agnivade) [inline server/go.mod:76]: Please revert any changes to go.mod and go.sum.

**Review by agnivade — CHANGES_REQUESTED** [review]

File: server/go.mod (hunk 1 above)
Comment (TealWater) [reply server/go.mod]: changes have been addressed.

**Review by agnivade — APPROVED** [review]

**Comment by TealWater** [comment]
Just an FYI, I will be submitting multiple PRs after this one since there are changes in over 100+ files.

They will be limited to about 10-20 files for ease of readability.

File: server/channels/app/slashcommands/command_invite_people.go [hunk 2]
```diff
@@ -16,15 +16,15 @@ type InvitePeopleProvider struct {
 }
 
 const (
-	CmdInvite_PEOPLE = "invite_people"
+	CmdInvitePEOPLE = "invite_people"
```
Comment (hanzei) [inline server/channels/app/slashcommands/command_invite_people.go:19]: Why not
```suggestion
	CmdInvitePeople = "invite_people"
```
?

**Review by hanzei — APPROVED** [review]
Nice one. :+1:  One non-blocking suggestion below.

**Comment by hanzei** [comment]
@TealWater Please let me know what you think about the suggestion. Then we can merge the PR.

File: server/channels/app/slashcommands/command_invite_people.go (hunk 2 above)
Comment (TealWater) [reply server/channels/app/slashcommands/command_invite_people.go:19]: This is an easy fix, I'll implement this.

Comment (TealWater) [reply server/channels/app/slashcommands/command_invite_people.go]: implemented.

**Comment by hanzei** [comment]
Thanks @TealWater !

**Pull request merged**